src/components/VideoProvider/useBackgroundSettings/useBackgroundSettings.ts

```diff
@@ -65,13 +68,40 @@ export const backgroundConfig = {
   images,
 };
 
-// TODO : Add video processing logic after backgroundSettings change
-// useEffect hooks, etc ...
+const virtualBackgroundAssets = '/virtualbackground';
 
-export default function useBackgroundSettings() {
-  const [backgroundSettings, setBackgroundSettings] = useState({
-    type: 'none',
-    index: 0,
-  } as BackgroundSettings);
-  return [backgroundSettings, setBackgroundSettings] as const;
+let blurProcessor: GaussianBlurBackgroundProcessor;
+blurProcessor = new GaussianBlurBackgroundProcessor({
+  assetsPath: virtualBackgroundAssets,
+});
+blurProcessor.loadModel();
+
+export default function useBackgroundSettings(videoTrack: LocalVideoTrack | undefined) {
+  const [backgroundSettings, setBackgroundSettings] = useState<BackgroundSettings>(() => {
+    const localStorageSettings = window.localStorage.getItem(SELECTED_BACKGROUND_SETTINGS_KEY);
+    return localStorageSettings ? JSON.parse(localStorageSettings) : { type: 'none', index: 0 };
+  });
+
+  // Specifically used for handling room disconnection
+  const removeProcessor = (): void => {
+    if (videoTrack && videoTrack.processor) {
+      videoTrack.removeProcessor(videoTrack.processor);
+    }
+  };
+
+  useEffect(() => {
+    if (videoTrack) {
+      if (videoTrack.processor) {
+        videoTrack.removeProcessor(videoTrack.processor);
+      }
```

Comment:
You can reuse the method you created above
```suggestion
     removeProcessor();
```

Reply:
Hey Charlie, I made it a separate function and didn't use it inside the useEffect hook since I was getting warnings about missing `removeProcessor` as a dependency, but when I include it, I get another warning about how `removeProcessor` is updated on every render. What would be the best step/approach to this?

Reply:
cc @timmydoza

Reply:
There's a way to make this work, but I wonder if we need to worry about removing the processor when we disconnect from a room?  When we disconnect from a room, track.stop() is automatically called because of this hook [here](https://github.com/twilio/twilio-video-app-react/blob/master/src/components/VideoProvider/useHandleRoomDisconnection/useHandleRoomDisconnection.ts#L22) (this is to turn off the camera light on laptops when the user disconnects from a room).  Then when the user goes back to the Join Room screens, new tracks are acquired.

@charliesantos Since we are stopping the tracks on disconnect, do we need to remove the processor as well?

If we do, I think we can just handle that in the `removeVideoTrack` function in the `useLocalTracks` hook [here](https://github.com/twilio/twilio-video-app-react/blob/master/src/components/VideoProvider/useLocalTracks/useLocalTracks.ts#L45). Something like this maybe (but only if needed):

```
  const removeLocalVideoTrack = useCallback(() => {
    if (videoTrack) {
      if (videoTrack.processor) {
        videoTrack.removeProcessor(videoTrack.processor)
      }
      videoTrack.stop();
      setVideoTrack(undefined);
    }
  }, [videoTrack]);
  ```

Reply:
Yes we need to remove the processor that way it won't show in the join screen. On @timmydoza 's suggestion, it doesn't seem to save us anything 😄 . I was hoping we don't have to duplicate the code. I think I'm leaning more towards keeping our current implementation. We will probably remove this later if we decide to launch the virtual background settings UI in the join screen page.